File: Problema1/Factura.java
class Factura {

    private String numeroPieza;
    private String descripcionPieza;
    private int cantidad;
    private double precioArticulo;

    public Factura(String numeroPieza, String descripcionPieza, int cantidad, double precioArticulo) {
        this.numeroPieza = numeroPieza;
        this.descripcionPieza = descripcionPieza;

        if (cantidad > 0) {
            this.cantidad = cantidad;
        } else {
            this.cantidad = 0;
        }

        if (precioArticulo > 0) {
            this.precioArticulo = precioArticulo;
        } else {
            this.precioArticulo = 0.0;
        }
    }

    public double obtenerMontoFactura() {
        return cantidad * precioArticulo;
    }
}

public class Main {

    public static void main(String[] args) {

        Factura factura1 = new Factura(
            "A001",
            "Martillo",
            3,
            150.0
        );

        System.out.println("Monto de la factura: $" + factura1.obtenerMontoFactura());
    }
}

File: Problema1/Factura.cpp
#include <iostream>
#include <string>

using namespace std;

class Factura
{
private:
    string numeroPieza;
    string descripcionPieza;
    int cantidad;
    double precioArticulo;

public:
    Factura(string numeroPieza, string descripcionPieza, int cantidad, double precioArticulo)
    {
        this->numeroPieza = numeroPieza;
        this->descripcionPieza = descripcionPieza;

        if (cantidad > 0)
            this->cantidad = cantidad;
        else
            this->cantidad = 0;

        if (precioArticulo > 0)
            this->precioArticulo = precioArticulo;
        else
            this->precioArticulo = 0.0;
    }

    double obtenerMontoFactura()
    {
        return cantidad * precioArticulo;
    }
};

int main()
{
    Factura factura1("A001", "Martillo", 3, 150.0);

    cout << "Monto de la factura: $"
         << factura1.obtenerMontoFactura()
         << endl;

    return 0;
}
